commands/scanpullrequest.go

```diff
@@ -39,9 +40,35 @@ func (cmd *ScanPullRequestCmd) Run(configAggregator utils.RepoAggregator, client
 			return err
 		}
 	}
+	if err := cmd.verifyDifferentBranches(repoConfig); err != nil {
+		return err
+	}
 	return scanPullRequest(repoConfig, client)
 }
 
+// Verifies current branch and target branch are not the same.
+// The Current branch is the branch the action is triggered on.
+// The Default branch is the branch to open pull request to.
+func (cmd *ScanPullRequestCmd) verifyDifferentBranches(repoConfig *utils.Repository) error {
```

Comment:
This function is great. I have the following suggestions though:
1. Can we run it in 'dry-run' mode only, so that we avoid this processing during real operation?
2. I suggest we add a test for it.

Reply:
Why wouldn't we want this on real operations? 
I want to fail the command instead of getting false results ( by scanning the same branch)

There is a test for that :
TestScanPullRequestSameBranchFail